.github/workflows/macos.yml

```diff
@@ -0,0 +1,142 @@
+
+jobs:
+  cmake:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        EVENT_MATRIX: ["NONE","DISABLE_OPENSSL","DISABLE_THREAD_SUPPORT","DISABLE_DEBUG_MODE","DISABLE_MM_REPLACEMENT","TEST_EXPORT_STATIC","TEST_EXPORT_SHARED","OPENSSL_1_1"]
+
+    steps:
+    - uses: actions/checkout@v2.0.0
+
+    - name: Cache
+      uses: actions/cache@v1.1.0
+      with:
+        path: build
+        key: ${{ matrix.os }}-cmake-${{ matrix.EVENT_MATRIX }}
+
+    - name: Build And Test
+      shell: bash
+      run: |
+        if [ "${{ matrix.EVENT_MATRIX }}" == "OPENSSL_1_1" ]; then
+          export OPENSSL_ROOT_DIR=/usr/local/opt/openssl@1.1
+        else
+          export OPENSSL_ROOT_DIR=/usr/local/opt/openssl
+        fi
+
+        if [ "${{ matrix.EVENT_MATRIX }}" == "DISABLE_OPENSSL" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__DISABLE_OPENSSL=ON"
+
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "DISABLE_THREAD_SUPPORT" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__DISABLE_THREAD_SUPPORT=ON"
+
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "DISABLE_DEBUG_MODE" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__DISABLE_DEBUG_MODE=ON"
+
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "DISABLE_MM_REPLACEMENT" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__DISABLE_MM_REPLACEMENT=ON"
+
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "TEST_EXPORT_STATIC" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__LIBRARY_TYPE=STATIC -DEVENT__DISABLE_TESTS=ON -DEVENT__DISABLE_SAMPLES=ON"
+
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "TEST_EXPORT_SHARED" ]; then
+          EVENT_CMAKE_OPTIONS="-DEVENT__LIBRARY_TYPE=SHARED -DEVENT__DISABLE_TESTS=ON -DEVENT__DISABLE_SAMPLES=ON"
+
+        else
+          EVENT_CMAKE_OPTIONS=""
+        fi
+
+        #run build and test
+        JOBS=20
+        export CTEST_PARALLEL_LEVEL=$JOBS
+        export CTEST_OUTPUT_ON_FAILURE=1
+        mkdir -p build
+        cd build
+        echo [cmake]: cmake .. $EVENT_CMAKE_OPTIONS
+        cmake .. $EVENT_CMAKE_OPTIONS
+        cmake --build .
+        if [ "${{ matrix.EVENT_MATRIX }}" == "TEST_EXPORT_STATIC" ]; then
+          sudo python3 ../test-export/test-export.py static
+        elif [ "${{ matrix.EVENT_MATRIX }}" == "TEST_EXPORT_SHARED" ]; then
+          sudo python3 ../test-export/test-export.py shared
+        else
+          cmake --build . --target verify
+        fi
+
+    - uses: actions/upload-artifact@v1
+      if: failure()
+      with:
+        name: ${{ matrix.os }}-cmake-${{ matrix.EVENT_MATRIX }}-build
+        path: build
+
+  autotools:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest]
+        EVENT_MATRIX: ["NONE","DISABLE_OPENSSL","DISABLE_THREAD_SUPPORT","DISABLE_DEBUG_MODE","DISABLE_MM_REPLACEMENT","OPENSSL_1_1"]
+
+    steps:
+    - uses: actions/checkout@v2.0.0
+    - name: Cache
+      uses: actions/cache@v1.1.0
```

Comment:
I'm seeing there is no cache for [some builds](https://github.com/ygj6/libevent/runs/400990964#step:3:5), does it saved if the build fails or it is just the first that build run?

P.S. cache greatly decreases build time, especially on win32

Reply:
Only first time and success actions can save the build folder

Reply:
So 

>and success actions can save the build folder

in case of failure the cache will not be stored and there is no ability to do so? (like `APPVEYOR_SAVE_CACHE_ON_ERROR`)

Reply:
There's not much we can do right now because we can't change the behavior of `actions/checkout` unless we write the cache code ourselves.